### PR TITLE
integrate npm package evm and remove dependency mythril cli, add opti…

### PR DESCRIPTION
**Pull request by thecodingshrimp**
…onal parameter code_size for a single tx

Pull request solving #17

### CODE REVIEW

**Review by informartin — CHANGES_REQUESTED** [review]
Looks like a good solution to me, especially as no external CLI calls are required anymore.

Currently, all addresses are retrieved from the bytecode and the respective contracts are migrated recursively. An address could also refer to an EOA though. Therefore, it should be checked on the source chain, if it is indeed a contract. 
For references stored in the state that is already checked by retrieving the contract code:
https://github.com/informartin/VeriSmart/blob/ec51807299ed0d67b382b7722dd2c21ea449149e/redeployContract.js#L37-L42

**Review by informartin — APPROVED** [review]

**Review by informartin — APPROVED** [review]

File: contracts/InitContract.json [hunk 1]
```diff
@@ -18,10 +30,19 @@
...
   "sourceMap": "57:233:0:-;;;;8:9:-1;5:2;;;30:1;27;20:12;5:2;57:233:0;;;;;;;",
   "deployedSourceMap": "57:233:0:-;;;;;;;;;;;;;;;;;;;;;;;;85:203;;8:9:-1;5:2;;;30:1;27;20:12;5:2;85:203:0;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;177:6;:13;162:4;:11;:28;154:37;;;;;;;;215:42;201:66;;;268:4;274:6;201:80;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;23:1:-1;8:100;33:3;30:1;27:10;8:100;;;99:1;94:3;90:11;84:18;80:1;75:3;71:11;64:39;52:2;49:1;45:10;40:15;;8:100;;;12:14;201:80:0;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;23:1:-1;8:100;33:3;30:1;27:10;8:100;;;99:1;94:3;90:11;84:18;80:1;75:3;71:11;64:39;52:2;49:1;45:10;40:15;;8:100;;;12:14;201:80:0;;;;;;;;;;;;;;;;;;;;;;;;8:9:-1;5:2;;;30:1;27;20:12;5:2;201:80:0;;;;8:9:-1;5:2;;;45:16;42:1;39;24:38;77:16;74:1;67:27;5:2;201:80:0;;;;85:203;;:::o",
   "source": "pragma solidity ^0.4.21;\n\nimport \"./ProxyContract.sol\";\n\ncontract InitContract {\n    function setValue(bytes32[] keys, bytes32[] values) public {\n        require(keys.length == values.length);\n        ProxyContract(0x2222222222222222222222222222222222222222).setValue(keys, values);\n    }\n}\n",
```
Comment (informartin) [inline contracts/InitContract.json:48]: Could you also add the source code of the changed contract, please?

Comment (thecodingshrimp) [reply contracts/InitContract.json:48]: EDIT: Unfortunately I don't get the code block for the contracts working in this comment.

As we discussed in a previous meeting you wanted to have it only in a json format as the source code is in the json file as well.

Below you'll find the updated versions of both contracts (I can also push them of course):

InitContract.sol:
`
pragma solidity ^0.4.21;

import "./ProxyContract.sol";

contract InitContract {

    function setValue(bytes32[] keys, bytes32[] values)
    public
    {
        require(keys.length == values.length);
        ProxyContract(0x2222222222222222222222222222222222222222).setValue(keys, values);
    }

    function close()
    public
    {
        emit Status('destructing');
        selfdestruct(0x0);
    }

    event Status(string indexed status);
}
`

ProxyContract.sol
`pragma solidity ^0.4.21;

contract ProxyContract {
    function setValue(bytes32[] keys, bytes32[] values) public {
        address initializationContract = 0x8888888888888888888888888888888888888888;
        require(msg.sender == initializationContract);
        require(keys.length == values.length);
        for(uint i = 0; i < keys.length; i++) {
            assembly {
                sstore(mload(add(add(keys, 0x20), mul(i, 0x20))), mload(add(add(values, 0x20), mul(i, 0x20))))
            }
        }
    }

    //https://blog.gnosis.pm/solidity-delegateproxy-contracts-e09957d0f201
    function() public payable {
        bool success = 0x2222222222222222222222222222222222222222.delegatecall(msg.data);
        assembly {
            let mempointer := mload(0x40)
            returndatacopy(mempointer, 0, returndatasize())
            switch success
                case 0 { revert(mempointer, returndatasize()) }
                default { return(mempointer, returndatasize()) }
        }
    }
}`

Comment (informartin) [reply contracts/InitContract.json:48]: > EDIT: Unfortunately I don't get the code block for the contracts working in this comment.

Could you elaborate, please? What code block isn't working?

> As we discussed in a previous meeting you wanted to have it only in a json format as the source code is in the json file as well.

Yes, I was thinking of adding the high level code under the `source` key of the json though.

Comment (informartin) [reply contracts/InitContract.json:48]: Alright, this is solved in PR #21.

**Review by informartin — APPROVED** [review]

**Pull request merged**